src/fastertransformer/utils/gemm_test/encoder_gemm_func.cc

```diff
@@ -543,4 +543,562 @@ template void generate_encoder_gemm_config<__nv_bfloat16>(
     int batch_size, int seq_len, int head_num, int size_per_head, void* buffer, bool isAppend);
 #endif
 
+
+template<typename T>
+void generate_effective_encoder_gemm_config(
```

Comment:
This function configures `m` from `1` to `max_seq_len`.  `generate_encoder_gemm_config` only considers `m = max_seq_len`.

Reply:
most of the code is same as above, not sure if we can do such change in generate_encoder_gemm_config, like add a switch there, default is not enable scanning, so it can be compatible with existing. 
Or call generate_encoder_gemm_config multiple times.

Reply:
this one covers `generate_encoder_gemm_config`. We can merge them. Currently, it is easier for debugging to separate them. Will merge them later.